agent/src/beerocks/slave/agent_db.h

```diff
@@ -72,8 +75,165 @@ class AgentDB {
+     * This function should be called only once with valid arguments. If called once and then called
+     * again, the radio struct on the database will not be updated with new arguments, and the
+     * function will return false.
+     * 
+     * @param front_iface_name Front iface name.
+     * @param back_iface_name Back iface name.
+     * @return true if a radio struct has been added to the database, otherwise return false.
+     */
+    bool add_radio(const std::string &front_iface_name, const std::string &back_iface_name);
+
+    /**
+     * @brief Get list of all radio objects on the database.
+     * This function shall be used in order to iterate over all the radios. 
+     * 
+     * @return const std::vector<std::string &>& Intefaces names list.
+     */
+    const std::vector<sRadio *> &get_radios_list() { return m_radios_list; };
+
+    /* Helper enum for get_mac_parent_radio() function */
+    enum class eMacType : uint8_t { ALL, RADIO, BSSID, CLIENT };
+
+    /**
+     * @brief Get a pointer to the parent radio struct of given MAC address. 
+     * 
+     * @param mac MAC address of radio interface or bssid.
+     * @return sRadio* A pointer to the radio struct containing the given MAC address.
+     */
+    sRadio *get_mac_parent_radio(const sMacAddr &mac, eMacType mac_type_hint = eMacType::ALL);
+
+    /**
+     * @brief Erase client from associated_clients list.
+     * 
+     * @param client_mac The client MAC address.
+     */
+    void erase_client(const sMacAddr &client_mac);
+
+    /**
+     * @brief 1905.1 Neighbor device information
+     * Information gathered from a neighbor device upon reception of a Topology Discovery message.
+     */
+    struct sNeighborDevice {
+        // MAC address of the interface on which the Topology Discovery message is transmitted.
+        sMacAddr transmitting_iface_mac;
+
+        // Timestamp of the last Topology Discovery message received from this neighbor device.
+        std::chrono::steady_clock::time_point timestamp;
+    };
+
+    /*
+     * @brief List of known 1905 neighbor devices.
+     * 
+     * Upper key: Local interface MAC on which the Topology Discovery message was received from.
+     * Upper value: Map containing 1905.1 device information ordered by neighbor device al_mac -
+     *  Sub-key: 1905.1 AL MAC address of the Topology Discovery message transmitting device.
+     *  Sub-value: 1905.1 device information.
+     * 
+     * Devices are being added to the list when receiving a 1905.1 Topology Discovery message from
+     * an unknown 1905.1 device. Every 1905.1 device shall send this message every 60 seconds, and
+     * we update the time stamp in which the message is received.
+     */
+    std::unordered_map<sMacAddr, std::unordered_map<sMacAddr, sNeighborDevice>> neighbor_devices;
+
+private:
+    std::list<sRadio> m_radios;
+    std::vector<sRadio *> m_radios_list;
```

Comment:
This is because I didn't want to give the user access to the radio container.
I wanted that adding of element will be done only with database API functions.
The second container is a helper container holding only the pointers to the radio elements on `m_radios`, used only by `get_radios_list()` which allows iterating on the container.